### PR TITLE
rm_brew_packages_add_xcode_check

### PR DESCRIPTION
**Pull request by BaselAl**
I discussed with Evstati about unnecessary brew formulae part of provision repo. Regarding our conversation I deleted several of them.

### CODE REVIEW

**Review by BaselAl — COMMENTED** [review]
I am sure you will mention that git clone automatically installes xcode but some of our new colleagues use "zip download" for downloading repo.

File: Developer.bundle [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #######################################
-# Developer file for host development machine
+# Developer file for host development machine #
```
Comment (hkdobrev) [inline Developer.bundle]: This formatting is strange. As per my comments in a previous PR, the `#` should be extended to cover the longer text. And then I suggested just to simplify the titles by having a single `#` in front with no top, bottom and right wrapping.

File: Developer.bundle [hunk 2]
```diff
@@ -22,16 +22,12 @@ brew 'brew-cask-completion'
 brew 'bundler-completion'
 brew 'gem-completion'
 brew 'open-completion'
-brew 'vagrant-completion'
```
Comment (hkdobrev) [inline Developer.bundle]: I don't think we should remove this while we have `vagrant` installed below.

File: Non-Developer.bundle [hunk 3]
```diff
@@ -1,5 +1,5 @@
 #######################################
-# Non-Developer file for host non-development machine
+# Non-Developer file for host non-development machine #
```
Comment (hkdobrev) [inline Non-Developer.bundle]: Same as previous heading. If not consistent, let's just simplify it, please.

File: Developer.bundle [hunk 4]
```diff
@@ -22,16 +22,12 @@ brew 'brew-cask-completion'
...
 
 # Latest versions of essential tools
 brew 'wget'
 brew 'curl'
 brew 'git'
 
-# PHP and tools
-brew 'php71'
```
Comment (hkdobrev) [inline Developer.bundle:33]: How would PHP be installed on the host? It's very useful for running Composer, other commands and even some unit tests on the host quickly. We could install PHP 7.3 now locally.

File: Developer.bundle [hunk 5]
```diff
@@ -40,19 +36,6 @@ brew 'rbenv'
 # JavaScript package manager
 brew 'yarn'
 
-# Image-related tools
-brew 'jpegoptim'
-brew 'optipng'
```
Comment (hkdobrev) [inline Developer.bundle:45]: Why remove these? They are very good image optimisers and we've even used them in some Automator scripts for the Catalogue production team.

**Review by hkdobrev — CHANGES_REQUESTED** [review]

**Comment by hkdobrev** [comment]
> git clone automatically installes xcode

I really don't think it does 😀 

`xcode-select --install` is something I'd recommend to everyone using macOS and compiling any formula with brew. It should be run after every macOS upgrade, actually.

I mean "Command Line Extensions".

**Comment by hkdobrev** [comment]
Also, `xcode-select --install` does not install XCode. It installs compiler dependencies and other dev tools which may be checked in Xcode preferences.

**Comment by hkdobrev** [comment]
The `.DS_Store` file shouldn't be committed in the repo. Please add and configure a useful global gitignore file to ignore platform-specific files in version control. See https://help.github.com/en/github/using-git/ignoring-files#configuring-ignored-files-for-all-repositories-on-your-computer and https://gist.github.com/octocat/9257657.

File: Developer.bundle (hunk 4 above)
Comment (BaselAl) [reply Developer.bundle:33]: Hi @hkdobrev Catalina comes with php 7.3 pre-installed.

Comment (hkdobrev) [reply Developer.bundle:33]: And macOS Big Sur comes with PHP 7.4. I wouldn't rely on any binary built-in with the OS like that. Especially if it's bundled with macOS. They might not compile with the official set of built-in extensions. They could either upgrade or not upgrade when you don't like. You don't get an easy way to update to new patch releases. You don't get a way to switch between PHP versions.

File: Developer.bundle (hunk 1 above)
Comment (BaselAl) [reply Developer.bundle]: @hkdobrev Strange from a programmatic point of view or in general ?
I would remove all # and just leave these in front of the text.

File: Developer.bundle (hunk 4 above)
Comment (BaselAl) [reply Developer.bundle:33]: @hkdobrev sure.

File: Developer.bundle (hunk 5 above)
Comment (BaselAl) [reply Developer.bundle:45]: @hkdobrev Who are they ? @EZarkov mentioned that no оne uses image optimisers. If someone wants to install image optimisers, there are no restrictions to do so.

**Comment by BaselAl** [comment]
> > git clone automatically installes xcode
> 
> I really don't think it does 😀
> 
> `xcode-select --install` is something I'd recommend to everyone using macOS and compiling any formula with brew. It should be run after every macOS upgrade, actually.

:D You are right but I mean git clone triggers an action to install command line tools or just type xcode-select --install.

File: Non-Developer.bundle (hunk 3 above)
Comment (BaselAl) [reply Non-Developer.bundle]: @hkdobrev Sure.

File: Developer.bundle (hunk 4 above)
Comment (BaselAl) [reply Developer.bundle:33]: type php --version on preinstalled Catalina and you will see PHP 7.3.11 (cli)
Catalina comes with php 7.3 pre-installed....
@hkdobrev The latest macOS 10.15 Catalina comes with Apache 2.4 pre-installed....
Screenshot 2020-06-30 at 1 22 06

File: Developer.bundle (hunk 2 above)
Comment (BaselAl) [reply Developer.bundle]: @hkdobrev okay

**Comment by BaselAl** [comment]
> The `.DS_Store` file shouldn't be committed in the repo. Please add and configure a useful global gitignore file to ignore platform-specific files in version control. See https://help.github.com/en/github/using-git/ignoring-files#configuring-ignored-files-for-all-repositories-on-your-computer and https://gist.github.com/octocat/9257657.

Sure. Thanks for the links.

File: Non-Developer.bundle (hunk 3 above)
Comment (hkdobrev) [reply Non-Developer.bundle]:
```suggestion
# Non-Developer file for host non-development machine
```

File: pecl_installs.sh [hunk 6]
```diff
@@ -1,4 +1,9 @@
 #/usr/bin/env bash
 
+# Installing PEAR
+cd /tmp
+curl -s -O https://pear.php.net/install-pear-nozlib.phar
+sudo php install-pear-nozlib.phar -d /usr/local/lib/php -b /usr/local/bi
+
 # Install an extension for PHP
-pecl install xdebug
+sudo pecl install xdebug
```
Comment (hkdobrev) [inline pecl_installs.sh]: Let's remove pecl and xdebug from the host as not needed there.

**Review by BaselAl — COMMENTED** [review]
Please check my changes.

**Review by BaselAl — COMMENTED** [review]
I am afraid but the name of the commit is cut off.

**Review by hkdobrev — APPROVED** [review]

**Review by pmilev — APPROVED** [review]

**Pull request merged**